docs/io/index_unofficial.rst

```diff
@@ -0,0 +1,283 @@
+==========================================================
+
+This class provides the high-level interface for reading data from each data
+format. Read about the purpose of EventFileReaders at
+:ref:`EventFileReaderClasses`.
+
+Location for every unofficial `ctapipe.io.eventfilereader.EventFileReader`.
+
+Each reader is required to import their
+:ref:`Low-Level Reader-Specific Modules` inside their `__init__` method,
+therefore bypassing the need for the external reader software to be installed
+unless someone has chosen to use this specific
+`ctapipe.io.eventfilereader.EventFileReader`, in which case a
+`ModuleNotFoundError` will be raised if the external software has not been
+found.
+
+The reader must also implement a fast method for
+`ctapipe.io.eventfilereader.EventFileReader.check_file_compatibility`. This
+method attempts to find a reader which *may* be compatible with the input
+file, and can be as simple as looking for a unique file extension in the file
+path or attempting to find a header keyword in the file. It is strongly
+advised that this method does not depend on the external reader software, as
+this would introduce a dependency when choosing the correct reader in the
+`ctapipe.io.eventfilereader.EventFileReaderFactory`.
+
+The currently existing EventFileReaders,
+`ctapipe.io.eventfilereader.HessioFileReader` and
+`ctapipe.io.unofficial.eventfilereader.TargetioFileReader` can be used as
+templates to create a new EventFileReader for a different data format.
+
+
+Low-Level Reader-Specific Modules
+===========================
+
+A seperate module should be created for the low-level method used for filling
+the event containers for a specific data format. The naming convention for
+this module is "\*io.py", where "\*" is the short-name for the data format.
+For example: `ctapipe.io.hessio` and `ctapipe.io.unofficial.targetio.targetio`.
+
+The low-level module can be a directory of multiple modules, allowing the
+various components involved in the event building (such as containers) to be
+grouped together for a particular data format.
+
+Each data format can be wildly different, therefore these low-level
+modules can appear very different in how they function. This is fine as long
+as these rules are followed:
+
+* The `ctapipe.io.eventfilereader.EventFileReader` calls the low-level methods
+  such that the high-level interface remains the same as other
+  `ctapipe.io.eventfilereader.EventFileReader`.
+* The low-level methods fill the containers with the relevant data, e.g. the
+  waveforms are stored in "data.r0.tel[].adc_samples" (assuming they are
+  waveforms of data level R0, see the next section for an explanation of data
+  levels and see `ctapipe.io.hessio` and
+  `ctapipe.io.unofficial.targetio.targetio` for examples)
+* The unique 'origin' string is added to the meta dictionary of the container
+  so the data source of the events can be obtained by other algorithms, e.g.
+
+  >>> data.meta['origin'] = "targetio"
```

Comment:
A rule is missing here, how the origin string should be created. Should it be the module name? Or the class-name of the reader who created this? 

And why should it be a string? What will this string be used for? If is it used for dynamically executing some code, based on the "origin", when instead of putting a string there, one might want to hook an function-, class- or module-reference there, so the dynamic executing becomes much easier and safer.

Reply:
Yeah, I don't think this is used anywhere yet, except debugging and testing that the events have the expected origin.

The string should just be the name of the data format.